syntax = "proto3";
package cosmos.bank.v1beta1;

import "gogo.proto";
import "coin.proto";
import "cosmos.proto";
import "msg.proto";
import "google/protobuf/descriptor.proto";

option go_package = "github.com/cosmos/cosmos-sdk/x/bank/types";


// MsgSend represents a message to send coins from one account to another.
message MsgSend {
  option (cosmos.msg.v1.signer) = "from_address";

  option (gogoproto.equal)           = false;
  option (gogoproto.goproto_getters) = false;

  string   from_address                    = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
  string   to_address                      = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
  repeated cosmos.base.v1beta1.Coin amount = 3
      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
}
